homeassistant/components/knx/config_flow.py

```diff
@@ -333,9 +346,16 @@ async def async_step_tunnel(
         }
 
         if user_input is not None:
+            connection_type = user_input[CONF_KNX_TUNNELING_TYPE]
             entry_data = {
                 **entry_data,
-                **user_input,
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_PORT: user_input[CONF_PORT],
+                ConnectionSchema.CONF_KNX_ROUTE_BACK: connection_type
+                == CONF_KNX_LABEL_TUNNELING_UDP_ROUTE_BACK,
+                CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING_TCP
+                if connection_type == CONF_KNX_LABEL_TUNNELING_TCP
+                else CONF_KNX_TUNNELING,
```

Comment:
```suggestion
                ConnectionSchema.CONF_KNX_ROUTE_BACK: (
                    connection_type == CONF_KNX_LABEL_TUNNELING_UDP_ROUTE_BACK
                ),
                CONF_KNX_CONNECTION_TYPE: (
                    CONF_KNX_TUNNELING_TCP
                    if connection_type == CONF_KNX_LABEL_TUNNELING_TCP
                    else CONF_KNX_TUNNELING
                ),
```
I think thats better readable.